pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -415,6 +416,54 @@ func buildFilterPreemptionCliArgs(
 	return args, nil
 }
 
+// GetHostErrorVMs checks the host error status of the given VMs, by querying the GCP logging service.
+func (p *Provider) GetHostErrorVMs(l *logger.Logger, since time.Time) ([]vm.PreemptedVM, error) {
+	args, err := buildFilterHostErrorCliArgs(p.GetProject(), since)
+	if err != nil {
+		l.Printf("Error building gcloud cli command: %v\n", err)
+		return nil, err
+	}
+	l.Printf("gcloud cli for host error : " + strings.Join(append([]string{"gcloud"}, args...), " "))
+	var logEntries []LogEntry
+	if err := runJSONCommand(args, &logEntries); err != nil {
+		l.Printf("Error running gcloud cli command: %v\n", err)
+		return nil, err
+	}
+	// Extract the VM name and the time of host error from logs.
+	var hostErrorVMs []vm.PreemptedVM
+	for _, logEntry := range logEntries {
+		timestamp, err := time.Parse(time.RFC3339, logEntry.ReceiveTimestamp)
+		if err != nil {
+			l.Printf("Error parsing gcp log timestamp, Preemption time not available: %v", err)
+			hostErrorVMs = append(hostErrorVMs, vm.PreemptedVM{Name: logEntry.ProtoPayload.ResourceName})
+			continue
+		}
+		hostErrorVMs = append(hostErrorVMs, vm.PreemptedVM{Name: logEntry.ProtoPayload.ResourceName, PreemptedAt: timestamp})
```

Comment:
It doesn't look like we are using the time of error at all right? I wonder if we even need a struct for this or if we can just return `[]string`. That would answer your other question of if you should reuse the `vmPreemption` struct or not.

Reply:
Updated